File: solutions/kotlin/bob/1/src/main/kotlin/Bob.kt
object Bob {
    fun hey(input: String) = when {
        input.isBlank() -> "Fine. Be that way!"
        input.yelling() && input.question() -> "Calm down, I know what I'm doing!"
        input.yelling() -> "Whoa, chill out!"
        input.question() -> "Sure."
        else -> "Whatever."
    }
}

private fun String.question() = if (this.isEmpty()) false else this.trim().last() == '?'

private fun String.yelling() = this == this.toUpperCase() && this.any { it.isLetter() }

File: solutions/typescript/bob/1/bob.ts
export function hey(message: string): string {
  const m = message.trim()
  const isYelling = m.match(/[A-Z]/g) && m === m.toUpperCase()
  const isQuestion = m[m.length - 1] === '?'

  if (m.length == 0) {
    return 'Fine. Be that way!'
  } else if (isQuestion && isYelling) {
      return 'Calm down, I know what I\'m doing!'
  } else if (isQuestion) {
    return 'Sure.'
  } else if (isYelling) {
    return 'Whoa, chill out!'
  }
  
  return 'Whatever.'
}
